plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -405,7 +405,16 @@ hoistExpr var t =
                     (PIR.Def var' (PIR.mkVar () var', PIR.Strict))
                     mempty
 
-                t' <- compileExpr t
+                CompileContext {ccOpts=profileOpts} <- ask
+                t' <- do
+                    if coProfile profileOpts==All then do
+                        let ty = PLC._varDeclType var'
+                            varName = T.pack $ show $ PLC._varDeclName var'
```

Comment:
I think you could pass in the `Name` and let `traceInside` decide what to do with it. Not a big deal in this instance, but it makes more sense to me that the user of `traceInside` knows it needs a `Name`, and it's only `traceInside` itself that knows that it's going to e.g. print it so it can stick in a string.